katharsis-core/src/main/java/io/katharsis/core/internal/resource/ResourceFieldImpl.java

```diff
@@ -143,8 +146,17 @@ public int hashCode() {
 	public ResourceInformation getParentResourceInformation() {
 		return parentResourceInformation;
 	}
-
+	
+	@Override
+	public ResourceFieldAccessor getAccessor(){
+		if(accessor == null){
```

Comment:
completely agree those constructors have grown out of proportions. I created a followup ticket https://github.com/katharsis-project/katharsis-framework/issues/388. Currently there are various PRs floating around and one more to come with all of them working in this area. So a refactoring in this area would currently lead to quite massive conflicts. I do that after the PRs are done.